docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1519 @@
+        pass
+
+Since ``tree.command()`` is more concise and easier to understand,
+it'll be the main method used to create slash commands in this guide.
+
+Some information is logically inferred from the function to populate the slash command's fields:
+
+- The :attr:`~.app_commands.Command.name` takes after the function name "meow"
+- The :attr:`~.app_commands.Command.description` takes after the docstring "Meow meow meow"
+
+To change them to something else, ``tree.command()`` takes ``name`` and ``description`` keyword-arguments:
+
+.. code-block:: python
+
+    @client.tree.command(name='woof', description='Woof woof woof')
+    async def meow(interaction: discord.Interaction):
+        pass
+
+    # or...
+    @client.tree.command(name='list')
+    async def list_(interaction: discord.Interaction):
+        # prevent shadowing the "list" builtin
+
+If a description isn't provided through ``description`` or by the docstring, an ellipsis "..." is used instead.
+
+Interaction
+++++++++++++
+
+As shown above, app commands always keep the first parameter for an :class:`~discord.Interaction`,
+a Discord model used for both app commands and UI message components.
+
+When an interaction is created on command invoke, some information about the surrounding context is given, such as:
+
+- :class:`discord.Interaction.channel` - the channel it was invoked in
+- :class:`discord.Interaction.guild` - the guild it was invoked in, if any
+- :class:`discord.Interaction.user` - the user or member who invoked the command
+
+Attributes like these and others are a given, however when it comes to responding to an interaction,
+by sending a message or otherwise, the methods from :attr:`.Interaction.response` need to be used.
+
+A response needs to occur within 3 seconds, otherwise this message pops up on Discord in red:
+
+.. image:: /images/guide/app_commands/interaction_failed.png
+
+In practice, it's common to use either of the following two methods:
+
+- :meth:`.InteractionResponse.send_message` to send a message
+- :meth:`.InteractionResponse.defer` to defer a response
+
+In the case of deferring, a follow-up message needs to be sent within 15 minutes for app commands.
+
+For example, to send a deferred ephemeral message:
+
+.. code-block:: python
+
+    import asyncio
+    import random
+
+    @client.tree.command()
+    async def weather(self, interaction: discord.Interaction):
+        await interaction.response.defer(ephemeral=True) # indicates the follow-up message will be ephemeral
+
+        weathers = ['sunny', 'clear', 'cloudy', 'rainy', 'stormy', 'snowy']
+        await asyncio.sleep(5) # an expensive operation... (no more than 15 minutes!)
+        forecast = random.choice(weathers)
+
+        await interaction.followup.send(f'the weather today is {forecast}!')
+
+Syncing
+++++++++
+
+In order for this command to show up on Discord, the API needs some information regarding it, namely:
+
+- The name and description
```

Comment:
Also, sync when you add or modify permissions (`default_permissions` and `nsfw`). It might be useful to denote when not to sync as well (pretty much if any of the code after the `async def ():` part, and library side checks)